src/components/Tooltip/Tooltip.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+  <div class="relative">
+    <div
+      :class="[
+        'absolute'
+      ]"
+      :style="tooltipPositionStyle"
+    >
+      <div
+        ref="tooltipContainer"
+        :class="[
+          'px-4 py-3 text-xs rounded-md m-auto bg-gray-700 text-white relative',
+          {'opacity-0': !hover}
+        ]"
+      >
+        <slot name="content" />
+        <div
+          :class="[
+            'absolute bg-transparent w-0 h-0 m-auto',
+            {'border-l-8 border-r-8 border-b-8 border-l-transparent border-r-transparent border-b-gray-700 -top-2' : hasUpArrow},
+            {'border-l-8 border-r-8 border-t-8 border-l-transparent border-r-transparent border-t-gray-700 -bottom-2' : hasDownArrow},
+            {'border-t-8 border-b-8 border-r-8 border-t-transparent border-b-transparent border-r-gray-700 -left-2' : hasLeftArrow},
+            {'border-t-8 border-b-8 border-l-8 border-t-transparent border-b-transparent border-l-gray-700 -right-2' : hasRightArrow},
+            {'top-0 bottom-0': arrowIsVerticallyCenter},
+            {'left-0 right-0': arrowIsHorizontallyCenter},
+            {'left-4': arrowIsLeftOfCenter},
+            {'right-4': arrowIsRightOfCenter},
+
+          ]"
+        />
+      </div>
+    </div>
+    <div
+      ref="triggerContainer"
+      @mouseover="handleMouseover"
+      @mouseleave="handleMouseleave"
+    >
+      <slot
+        name="trigger"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'Tooltip',
+  props: {
+    position: {
+      type: String,
+      default: 'bottom',
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['top', 'bottom', 'left', 'right'].indexOf(value) !== -1;
+      }
+    },
+    arrowPlacement: {
+      type: String,
+      default: '',
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['', 'left', 'right'].indexOf(value) !== -1;
```

Comment:
Would it make sense to change `''` to `default` or `center` and have it default to that? For consistency sake. (e.g. when I went to test in Storybook, I was confused for a hot second on how I would force it back to the center) I don't feel strongly about this either way, just a suggestion.

Reply:
Will do!